Packages/io.chainsafe.web3-unity/Runtime/Scripts/Web3Accessor.cs

```diff
@@ -46,8 +47,9 @@ public static async Task TerminateAndClear(bool logout = false)
             {
                 Debug.LogError("Instance was not set.");
             }
-
+            LoadingOverlay.ShowLoadingOverlay("Logging out...");
             await Instance.web3.TerminateAsync(logout);
+            LoadingOverlay.HideLoadingOverlay();
```

Comment:
I think we need to use finally statement each time we're using Show/HideLoadingOverlay(). This way it won't get stuck if TerminateAsync throws an exception.

Reply:
Or a custom using statement perhaps?

Reply:
Ideally we should make loading overlay only show when used inside the custom `using` statement to enforce the 'safe' usage.

Reply:
I mean that you can create a class, return an instance of this class in `ShowOverlay`, and then use the instance as an argument for `using (instance)` operator. This way when execution gets out of the using statement block (organically or cause of the exception) it will call `Dispose` on the instance.

The instance can be initialized with the reference of the overlay and will be able to call `HideOverlay` in `Dispose`. If you put the class of the instance in the Overlay class, it will even be able to call private methods of the overlay.

You can't really enforce using the `using` statement this way, but you can make it an obvious choice, idk

Reply:
Ah I see! 
But the loading overlay itself is a classic unitys singleton object, so I didn't want to enforce such use-cases for something as simple as loadingoverlay is.

Reply:
Well, as I see it. It's either **using with disposable** or you have to cover the 3 lines in **`try-finally`** each time you want to show a loading overlay. If not it will keep getting stuck.

Reply:
Yeah the latter seems better for now. We're not going to use it that extensively tho, only in two places basically.